datalake/src/main/java/com/sequenceiq/datalake/service/FreeipaService.java

```diff
@@ -18,7 +19,17 @@
     @Inject
     private FreeIpaV1Endpoint freeIpaV1Endpoint;
 
-    public DescribeFreeIpaResponse describe(String envCrn) {
+    public void checkFreeipaRunning(String envCrn) {
+        DescribeFreeIpaResponse freeipa = describe(envCrn);
+        if (freeipa != null && freeipa.getAvailabilityStatus() != null && !freeipa.getAvailabilityStatus().isAvailable()) {
```

Comment:
I don't get it. If the freeipa is null or the availability status is null then we don't throw exception? Can we throw an exception in those cases as well and catch the problem early on? Now it doesn't really fail fast.

Reply:
This is exactly the former logic present in 2 services in method `private void checkFreeipaRunning(String envCrn) {` extracted to its service. Otherwise, your suggestion is valid, so I am addressing it.